### PR TITLE
Simplifiction of OpenGL GUI shaders

### PR DESCRIPTION
**Pull request by sarbes**
## Description
This PR restructures code related to the loading of GUI GLSL shaders. Fragment shaders get merged into one file to reduce clutter.

## Motivation and Context
The main reason of this PR is to prepare the addition of shader effects for textures.

## How Has This Been Tested?
The code affects all shaders, excluding the video rendering.

I've tested both GLSL version 120 and 150 shaders on my PC (Manjaro, Mesa 20.0.6, Xorg, with a Polaris 10 - OpenGL 4.6) and couldn't spot any shortcomings. Older hardware/drivers might react differently to the changes in the fragment shader, so larger testing would be beneficial.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/rendering/gl/RenderSystemGL.cpp [hunk 1]
```diff
@@ -599,97 +599,44 @@ void CRenderSystemGL::InitialiseShaders()
...
 
-  m_pShader[SM_TEXTURE_NOBLEND].reset(new CGLShader("gl_shader_frag_texture_noblend.glsl", defines));
-  if (!m_pShader[SM_TEXTURE_NOBLEND]->CompileAndLink())
+  for (int i = SM_DEFAULT; i < SM_MAX; i++)
   {
-    m_pShader[SM_TEXTURE_NOBLEND]->Free();
-    m_pShader[SM_TEXTURE_NOBLEND].reset();
-    CLog::Log(LOGERROR, "GUI Shader gl_shader_frag_texture_noblend.glsl - compile and link failed");
-  }
+    switch(i)
```
Comment (Rechi) [inline xbmc/rendering/gl/RenderSystemGL.cpp:605]: > Control statement keywords have to be separated from opening parentheses by one space.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#34-whitespace

File: xbmc/rendering/gl/RenderSystemGL.cpp [hunk 2]
```diff
@@ -599,97 +599,44 @@ void CRenderSystemGL::InitialiseShaders()
...
+    {
+      case SM_DEFAULT:          defines += "#define SM_DEFAULT 1\n";          break;
```
Comment (Rechi) [inline xbmc/rendering/gl/RenderSystemGL.cpp:607]: > Do not put multiple statements on a single line. Always use a new line for a new statement.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#41-multiple-statements

File: xbmc/rendering/gl/RenderSystemGL.cpp [hunk 3]
```diff
@@ -599,97 +599,44 @@ void CRenderSystemGL::InitialiseShaders()
...
+      case SM_TEXTURE:          defines += "#define SM_TEXTURE 1\n";          break;
```
Comment (Rechi) [inline xbmc/rendering/gl/RenderSystemGL.cpp:608]: > Do not put multiple statements on a single line. Always use a new line for a new statement.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#41-multiple-statements

File: xbmc/rendering/gl/RenderSystemGL.cpp [hunk 4]
```diff
@@ -599,97 +599,44 @@ void CRenderSystemGL::InitialiseShaders()
...
+      case SM_TEXTURE_LIM:      defines += "#define SM_TEXTURE_LIM 1\n";      break;
```
Comment (Rechi) [inline xbmc/rendering/gl/RenderSystemGL.cpp:609]: > Do not put multiple statements on a single line. Always use a new line for a new statement.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#41-multiple-statements

File: xbmc/rendering/gl/RenderSystemGL.cpp [hunk 5]
```diff
@@ -599,97 +599,44 @@ void CRenderSystemGL::InitialiseShaders()
...
+      case SM_MULTI:            defines += "#define SM_MULTI 1\n";            break;
```
Comment (Rechi) [inline xbmc/rendering/gl/RenderSystemGL.cpp:610]: > Do not put multiple statements on a single line. Always use a new line for a new statement.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#41-multiple-statements

File: xbmc/rendering/gl/RenderSystemGL.cpp [hunk 6]
```diff
@@ -599,97 +599,44 @@ void CRenderSystemGL::InitialiseShaders()
...
+      case SM_FONTS:            defines += "#define SM_FONTS 1\n";            break;
```
Comment (Rechi) [inline xbmc/rendering/gl/RenderSystemGL.cpp:611]: > Do not put multiple statements on a single line. Always use a new line for a new statement.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#41-multiple-statements

File: xbmc/rendering/gl/RenderSystemGL.cpp [hunk 7]
```diff
@@ -599,97 +599,44 @@ void CRenderSystemGL::InitialiseShaders()
...
+      case SM_TEXTURE_NOBLEND:  defines += "#define SM_TEXTURE_NOBLEND 1\n";  break;
```
Comment (Rechi) [inline xbmc/rendering/gl/RenderSystemGL.cpp:612]: > Do not put multiple statements on a single line. Always use a new line for a new statement.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#41-multiple-statements

File: xbmc/rendering/gl/RenderSystemGL.cpp [hunk 8]
```diff
@@ -599,97 +599,44 @@ void CRenderSystemGL::InitialiseShaders()
...
+      case SM_MULTI_BLENDCOLOR: defines += "#define SM_MULTI_BLENDCOLOR 1\n"; break;
```
Comment (Rechi) [inline xbmc/rendering/gl/RenderSystemGL.cpp:613]: > Do not put multiple statements on a single line. Always use a new line for a new statement.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#41-multiple-statements

File: xbmc/rendering/gl/RenderSystemGL.cpp [hunk 9]
```diff
@@ -599,97 +599,44 @@ void CRenderSystemGL::InitialiseShaders()
...
+    }
+
```
Comment (Rechi) [inline xbmc/rendering/gl/RenderSystemGL.cpp:615]: remove the white space at the end of the line

**Review by Rechi — COMMENTED** [review]
Please follow the [current code guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md).

**Comment by sarbes** [comment]
I'll continue working on this after the launch of Matrix.

**Comment by sarbes** [comment]
I'll put this to rest. Thanks everybody.

**Pull request closed without merging**